rakelib/schemas.rake

```diff
@@ -236,22 +236,26 @@ def to_schema(data)
     key.to_s.start_with?('_')
   end.map(&:to_h)
 
+  # Flip plugin to true if the definition allows a plugin reference.
+  # This will allow plugin references for all sub-options and items.
+  plugin = metadata[:_plugin] if metadata[:_plugin]
+
   # Recurse through some properties, as each can have their own defintions.
-  %i[items additionalProperties propertyNames].each do |key|
+  %i[items additionalProperties].each do |key|
     next unless data.key?(key)
-    data[key] = to_schema(data[key])
+    data[key] = to_schema(data[key], plugin)
```

Comment:
This should use the `_plugin` definition for the suboption if it's defined, otherwise default to `_plugin` for the parent key.